### DISCUSSION TITLE
Feature Request: Service to enable/disable MPPT multi peak scanning

### DISCUSSION THREAD

**Discussion by michi1984** [wlcrs/huawei_solar#439] [Ideas]
Hi,

I would like to be able to enable and disable the MPPT multi peak scanning of my Huawei inverter via a service. This would be to optimise the use of the global peak MPP scan in scenarios where the PV-array is affected by shading only during 1-2 hours a day, depending on the season. Disabling the scan during the rest of the day where there is no shading could potentially slighly increase electricitiy production, because the energy loss during scanning is avoided. 

Currently this is only possible via the fusionsolar portal or the app (with direct wifi connection to the inverter), which is not really feasible to do. I would like to set up an automation which depending on the current situation switches the scan on and off.

This can be done via Modbus:
Signal Name: MPPT multimodal scanning 
Read and write: RW
Type: E16
Address: 42054
Number of Registers: 1

0 is off, 1 is on.

**Comment by wlcrs** [comment]
This register is not documented in the latest Modbus register definitions file? Can you please provide a source ?

**Reply by michi1984** [reply]
It is in the Modbus definitions for Sun2000MA. On Page 79 in the attached PDF.
[Modbus Interface Definitions Sun2000MA.pdf](https://github.com/wlcrs/huawei_solar/files/12313610/Modbus.Interface.Definitions.Sun2000MA.pdf)

**Comment by FLP-home** [comment]
@wlcrs Any planned actions in the upcoming releases to integrate the feature to control the MPPT scan (on/off) and the scan interval if it's activated?
Thanks a lot for your time in advance and this great open source HA integration!
